### PR TITLE
fix(qwik): dont pull in `node-fetch` in bundles

### PR DESCRIPTION
**Pull request by mhevery**
## Description

Add a short description of what changes you made, why you made them, and any other context that you think might be helpful for someone to better understand what is contained in this pull request. This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

_Screenshot_
If relevant, add a screenshot or two of the changes you made.

### CODE REVIEW

File: packages/sdks/src/functions/get-fetch.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
 import { getGlobalThis } from './get-global-this.js';
 
+/**
+ * The issue is that when building standalone bundle the bundler includes `node-fetch` as
```
Comment (samijaber) [inline packages/sdks/src/functions/get-fetch.ts:4]: which bundler are we talking about here? Is this a qwik optimizer/bundler issue in particular? or a Vite issue? can we add that information for clarity?

Comment (mhevery) [reply packages/sdks/src/functions/get-fetch.ts]: clarified

**Review by samijaber — APPROVED** [review]
Thanks for clarifying. To make sure I understand though: the "bundler" in this case is the Vite bundler used when we build the qwik SDK?

Also, was this causing any crashes/issues, or is it purely a bundle size optimization?

**Review by samijaber — CHANGES_REQUESTED** [review]
Actually, I'm not sure this hack is necessary. Instead, we need to exclude node modules from our vite bundles as a cleaner solution to this problem

**Comment by samijaber** [comment]
See how we do it for our Vue2 SDK:

https://github.com/BuilderIO/builder/blob/74f8c6d49b044aa52a43c3970bbaf5a5cf9bd30d/packages/sdks/output/vue/vue2/vite.config.ts#L14-L18

File: packages/sdks/src/functions/get-fetch.ts [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
-    const nodeFetch = import('node-fetch').then((d) => d.default) as Promise<
+    const nodeFetch = import(`node-fetch`).then((d) => d.default) as Promise<
       typeof global.fetch
     >;
-    return (nodeFetch as any).default || nodeFetch;
+    return (fetch = (nodeFetch as any).default || nodeFetch);
   }
 
-  return (globalFetch as any).default || globalFetch;
+  return (fetch = (globalFetch as any).default || globalFetch);
```
Comment (samijaber) [inline packages/sdks/src/functions/get-fetch.ts:35]: can we undo all the changes to this file now?

Comment (mhevery) [reply packages/sdks/src/functions/get-fetch.ts:16]: done

File: packages/sdks/src/functions/get-fetch.ts [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
```
Comment (samijaber) [inline packages/sdks/src/functions/get-fetch.ts:16]: Why is this new mutable `fetch` needed? I don't fully understand

Comment (samijaber) [reply packages/sdks/src/functions/get-fetch.ts:16]: (what happens if we don't have it in here)

**Review by samijaber — APPROVED** [review]

**Pull request merged**